examples/httpx_client.py

```diff
@@ -9,8 +9,10 @@
 from typing import AsyncIterator, Deque, Dict, Optional, Tuple, cast
 from urllib.parse import urlparse
 
-import httpcore
+import httpx
 from httpx import AsyncClient
+from httpx._transports.default import AsyncResponseStream
+from httpx._models import Request, Response
```

Comment:
This looks as though we're depending on private API, which is likely to break at any time or am I mistaken?

Reply:
Yes,well caught @jlaine.
You should import from the `httpx` rather than the (private) module space.

You want `httpx.Request` and `httpx.Response`.

The `AsyncResponseStream` class is private. Instead you'll want your own response stream that subclasses `httpx.AsyncByteStream()`...

```python
class AsyncResponseStream(AsyncByteStream):
    def __init__(self, response_stream: typing.AsyncIterable[bytes]):
        self._response_stream = response_stream

    async def __aiter__(self) -> typing.AsyncIterator[bytes]:
        # you might want to wrap any exceptions you can throw and coerce them into `httpx` exceptions.
        async for part in self._response_stream:
            yield part

    async def aclose(self) -> None:
        ... # hook into any close behaviour you have
```